lib/core/src/data/weakset.rs

```diff
@@ -0,0 +1,113 @@
+use std::collections::HashSet;
```

Comment:
Wait, after reviewing this file I realized that are not using `WeakSet` anywhere in you codebase, so why this file is committed? :D

Please read and reply to the reviews though.

Reply:
My mistake! I considered using it, but then I relied on the existing opt_vec implementation. Should we keep it?